src/status.rs

```diff
@@ -101,103 +101,124 @@ pub async fn serve(
+                            None => {
+                                info!(target: "lazymc", "Kicked player because lockout is enabled")
+                            }
+                        }
+                        action::kick(&client, &config.lockout.message, &mut writer).await?;
+                        break;
+                    }
+
+                    // Kick if client is banned
+                    if let Some(ban) = server.ban_entry(&client.peer.ip()).await {
+                        if ban.is_banned() {
+                            let msg = if let Some(reason) = ban.reason {
+                                info!(target: "lazymc", "Login from banned IP {} ({}), disconnecting", client.peer.ip(), &reason);
+                                reason.to_string()
+                            } else {
+                                info!(target: "lazymc", "Login from banned IP {}, disconnecting", client.peer.ip());
+                                DEFAULT_BAN_REASON.to_string()
+                            };
+                            action::kick(
+                                &client,
+                                &format!("{}{}", BAN_MESSAGE_PREFIX, msg),
+                                &mut writer,
+                            )
+                            .await?;
+                            break;
+                        }
+                    }
+
+                    // Kick if client is not whitelisted to wake server
+                    if let Some(ref username) = username {
+                        if !server.is_whitelisted(username).await {
+                            info!(target: "lazymc", "User '{}' tried to wake server but is not whitelisted, disconnecting", username);
+                            action::kick(&client, WHITELIST_MESSAGE, &mut writer).await?;
+                            break;
+                        }
+                    }
+
+                    // Start server if not starting yet
+                    Server::start(config.clone(), server.clone(), username).await;
+
+                    // Remember inbound packets
+                    inbound_history.extend(&raw);
+                    inbound_history.extend(&buf);
+
+                    // Build inbound packet queue with everything from login start (including this)
+                    let mut login_queue = BytesMut::with_capacity(raw.len() + buf.len());
+                    login_queue.extend(&raw);
+                    login_queue.extend(&buf);
+
+                    // Buf is fully consumed here
+                    buf.clear();
+
+                    // Route connection through proper channel
+                    if server.state() == server::State::Started {
+                        crate::service::server::route_proxy_queue(
+                            inbound,
+                            config.clone(),
+                            inbound_history.clone(),
+                        );
+                    } else {
```

Comment:
Not sure if this is right architecturally, tried to replicate the old functionality of
```
if server is started
then proxy
if not then status (occupy)
```